js_modules/dagit/packages/core/src/partitions/BackfillSelector.tsx

```diff
@@ -154,7 +154,9 @@ export const BackfillPartitionSelector: React.FC<{
     (name: string) => partitionData[name] === RunStatus.SUCCESS,
   );
   const setSelected = (selection: string[]) => {
-    _setSelected(selection);
+    const selectionSet = new Set(selection);
+    // first order the partition names
+    _setSelected(partitionNames.filter((name) => selectionSet.has(name)));
```

Comment:
It's a little unintuitive how this is sorting the selection - maybe also include a note that `partitionNames` is already sorted?

Reply:
hmm... yeah.  It's not sorted by any sort of sort order.   Partition names are given in a particular order by user code.  So we have to maintain the order that they're given to us... but yeah, can add a comment about it.